compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -12975,165 +12987,277 @@ private boolean isEndOfMappingBindingPattern(SyntaxKind nextTokenKind) {
+        STNode openParenthesis = parseOpenParenthesis(ParserRuleContext.OPEN_PARENTHESIS);
+        STNode argListBindingPatterns = parseErrorArgListBindingPatterns();
+        STNode closeParenthesis = parseCloseParenthesis();
+        endContext();
+        return STNodeFactory.createErrorBindingPatternNode(errorKeyword, typeRef, openParenthesis,
+                argListBindingPatterns, closeParenthesis);
     }
 
     /**
-     * Parse functional binding pattern.
+     * Parse error arg list binding pattern.
      * <p>
      * <code>
-     * functional-binding-pattern := functionally-constructible-type-reference ( arg-list-binding-pattern )
+     * error-arg-list-binding-pattern :=
+     * error-message-binding-pattern [, error-cause-binding-pattern] [, error-field-binding-patterns]
+     * | [error-field-binding-patterns]
      * <br/><br/>
-     * functionally-constructible-type-reference := error | type-reference
+     *
+     * error-message-binding-pattern := simple-binding-pattern
+     * <br/><br/>
+     *
+     * error-cause-binding-pattern := simple-binding-pattern | error-binding-pattern
+     * <br/><br/>
+     *
+     * simple-binding-pattern := capture-binding-pattern | wildcard-binding-pattern
+     * <br/><br/>
+     *
+     * error-field-binding-patterns :=
+     * named-arg-binding-pattern (, named-arg-binding-pattern)* [, rest-binding-pattern]
+     * | rest-binding-pattern
+     * <br/><br/>
+     *
+     * named-arg-binding-pattern := arg-name = binding-pattern
      * </code>
-     * 
-     * @param typeDesc Functionally constructible type reference
-     * @return Functional binding pattern node.
+     *
+     * @return Error arg list binding patterns.
      */
-    private STNode parseFunctionalBindingPattern(STNode typeDesc) {
-        STNode openParenthesis = parseOpenParenthesis(ParserRuleContext.ARG_LIST_START);
-        STNode argListBindingPatterns = parseArgListBindingPatterns();
-        STNode closeParenthesis = parseCloseParenthesis();
-        endContext();
-        return STNodeFactory.createFunctionalBindingPatternNode(typeDesc, openParenthesis, argListBindingPatterns,
-                closeParenthesis);
-    }
-
-    private STNode parseArgListBindingPatterns() {
+    private STNode parseErrorArgListBindingPatterns() {
         List<STNode> argListBindingPatterns = new ArrayList<>();
         SyntaxKind lastValidArgKind = SyntaxKind.CAPTURE_BINDING_PATTERN;
         STToken nextToken = peek();
 
         if (isEndOfParametersList(nextToken.kind)) {
             return STNodeFactory.createNodeList(argListBindingPatterns);
         }
-        argListBindingPatterns.add(parseArgBindingPattern());
 
-        nextToken = peek();
-        while (!isEndOfParametersList(nextToken.kind)) {
-            STNode argEnd = parseArgsBindingPatternEnd();
-            if (argEnd == null) {
-                // null marks the end of args
-                break;
-            }
+        int argCount = 1;
```

Comment:
Can we only rely on the `lastValidArgKind`?
I feel we can do all the validation only using the lastValidArgKind..